src/js/disability-benefits/686/config/form.js

```diff
@@ -1,40 +1,258 @@
-// import fullSchema from 'vets-json-schema/dist/686-schema.json';
+import { createSelector } from 'reselect';
+import fullSchema from 'vets-json-schema/dist/21-686C-schema.json';
+import _ from 'lodash/fp';
+
+import currentOrPastDateUI from '../../../common/schemaform/definitions/currentOrPastDate';
+import ssnUI from '../../../common/schemaform/definitions/ssn';
+import * as address from '../../../common/schemaform/definitions/address';
+import fullNameUI from '../../../common/schemaform/definitions/fullName';
 
 import IntroductionPage from '../containers/IntroductionPage';
 import ConfirmationPage from '../containers/ConfirmationPage';
+import SpouseMarriageView from '../components/SpouseMarriageView';
+
+import { validateAfterMarriageDate } from '../validation';
+
+const {
+  spouseDateOfBirth,
+  spouseSocialSecurityNumber,
+  spouseVaFileNumber,
+  liveWithSpouse,
+  spouseIsVeteran,
+} = fullSchema.properties;
 
-// const { } = fullSchema.properties;
+const {
+  marriages,
+  fullName,
+  ssn,
+  date,
+  vaFileNumber
+} = fullSchema.definitions;
 
-// const { } = fullSchema.definitions;
+function isMarried(form = {}) {
+  return ['Married', 'Separated'].includes(form.maritalStatus);
+}
+
+function createSpouseLabelSelector(nameTemplate) {
+  return createSelector(form => {
```

Comment:
I'm not 100% sure how reselect works but I think redefining the selector (first argument passed to `createSelector`) creates a new cache for each call to `createSpouseLabelSelector`. 

I think this could make this first argument its own selector (e.g. `spouseFullNameSelector`) by making its `resultFunc` an identity function. This would memoize the `spouseFullNameSelector` once (as long as the form is immutable) for the subsequent calls. 

Then `createSpouseLabelSelector` could pass `spouseFullNameSelector` as the first argument to its `createSelector` and the new `nameTemplate` callback for each of the different `updateSchema` `resultFunc`s.

Reply:
Oh, that's interesting. As far as I know, you're right about that being how reselect works, I was just focused on using reselect to keep data changes from forcing a new schema object because of the label each time. Reusing the caches across all these selectors seems like a good improvement.